src/components/FileUploaderV2/FileUploaderV2.js

```diff
@@ -0,0 +1,240 @@
+          {labelText}
+        </label>
+        <input
+          className="bx--visually-hidden"
+          ref={input => (this.input = input)}
+          id={this.uid}
+          type="file"
+          multiple={multiple}
+          accept={accept}
+          name={name}
+          onChange={onChange}
+          onClick={evt => {
+            evt.target.value = null;
+          }}
+        />
+      </div>
+    );
+  }
+}
+
+export function FilenameV2({
+  iconDescription,
+  onKeyDown,
+  status,
+  style,
+  tabIndex,
+  ...other
+}) {
+  switch (status) {
+    case 'uploading':
+      return (
+        <div
+          className="bx--loading"
+          style={{ ...style }}
+          tabIndex={tabIndex}
+          onKeyDown={onKeyDown}
+          role="button"
+          {...other}>
+          <svg className="bx--loading__svg" viewBox="-42 -42 84 84">
+            <circle cx="0" cy="0" r="37.5" />
+          </svg>
+        </div>
+      );
+    case 'edit':
+      return (
+        <Icon
+          description={iconDescription}
+          className="bx--file-close"
+          icon={iconCloseSolid}
+          style={style}
+          tabIndex={tabIndex}
+          onKeyDown={onKeyDown}
+          role="button"
+          {...other}
+        />
+      );
+    case 'complete':
+      return (
+        <Icon
+          description={iconDescription}
+          className="bx--file-complete"
+          icon={iconCheckmarkSolid}
+          style={style}
+          tabIndex={tabIndex}
+          onKeyDown={onKeyDown}
+          role="button"
+          {...other}
+        />
+      );
+    default:
+      return null;
+  }
+}
+FilenameV2.propTypes = {
+  onKeyDown: PropTypes.func,
+  style: PropTypes.object,
+  status: PropTypes.oneOf(['edit', 'complete', 'uploading']),
+  tabIndex: PropTypes.number,
+};
+FilenameV2.defaultProps = {
+  onKeyDown: () => {},
+  status: 'uploading',
+  style: {},
+  tabIndex: 0,
+};
+
+export default function FileUploaderV2(props) {
+  const handleClick = ({ evt, index }) => props.onClick({ evt, index });
+  const {
+    iconDescription,
+    buttonLabel,
+    buttonKind,
+    labelDescription,
+    labelTitle,
+    className,
+    multiple,
+    accept,
+    name,
+    onChange,
+    files,
+    ...other
+  } = props;
+
+  const classes = classNames({
+    'bx--form-item': true,
+    [className]: className,
+  });
+
+  return (
+    <div className={classes} {...other}>
+      <strong className="bx--label">{labelTitle}</strong>
+      <p className="bx--label-description">{labelDescription}</p>
+      <FileUploaderButtonV2
+        labelText={buttonLabel}
+        multiple={multiple || undefined}
+        buttonKind={buttonKind}
+        onChange={onChange}
+        accept={accept}
+        name={name}
+      />
+      <div className="bx--file-container">
+        {files.length
+          ? files.map((file, index) => (
+              <span key={uid()} className="bx--file__selected-file" {...other}>
```

Comment:
Instead of re-generating `key` for every render (will cause React-level re-rendering), can we assign a unique ID for each `file`?

Reply:
@asudoh would setting `key={file.uuid}` be a solution to this issue?

Reply:
@asudoh - Do you have an answer to this question? Looks like one of the last things we need to do to resolve.

Reply:
@asudoh thank you for reviewing! and yes @shinytoyrobots as @asudoh mentioned, I will be wrapping up work on this component according to the TODO list in the description of this PR, which may change as needed